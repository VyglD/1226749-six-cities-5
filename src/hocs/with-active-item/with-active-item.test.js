import React from "react";
import PropTypes from "prop-types";
import renderer from "react-test-renderer";
import withActiveItem from "./with-active-item";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveItem(MockComponent);

it(`Render correctly hoc withActiveItem`, () => {
  const component = renderer.create((
    <MockComponentWrapped>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(component).toMatchSnapshot();
});
