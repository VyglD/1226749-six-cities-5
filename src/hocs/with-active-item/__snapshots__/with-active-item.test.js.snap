// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly hoc withActiveItem 1`] = `<div />`;
